test/support/test_helpers.ex

```diff
@@ -0,0 +1,26 @@
+defmodule SingForNeeds.TestHelpers do
+    @moduledoc false
+    alias SingForNeeds.Artists.Artist
+    alias SingForNeeds.Repo
+
+  def artists_fixture do
+    artist1 =
+      %Artist{
+        name: "Artist 1",
+      }
+    Repo.insert!(artist1)
+
+    artist2 =
+      %Artist{
+        name: "Artist 2",
+      }
+    Repo.insert!(artist2)
+
+    artist3 =
+      %Artist{
+        name: "Artist 3",
+      }
+    Repo.insert!(artist3)
+    [artist1, artist2, artist3]
```

Comment:
For the `artists_fixture`, you're returning the `[artist1, artist2, artist3]` variables as you have them defined, not what is actually returned from `Repo.insert!`.

Each artist variable should have what's returned from the `Repo.insert!`, which, I believe, should be an `:ok` or `:error` tuple with an `%Artists{}` struct with all its new insertion information.

Do an `IO.inspect([artist1, artist2, artist3], label: "++++++")` to verify you're returning the correct version.

Reply:
Thanks @FedericoEsparza for this. @SherSpock and I looked into these and made the changes so as to return exactly what's inserted in the database. The `insert!` method however does not return the `{:ok, %Artist{}}` tuple (is normally returned with `insert` without the bang) but just the `%Artists{}` struct with whatever value was inserted.